contracts/lens/Quoter.sol

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+import "forge-std/console2.sol";
+import {Hooks} from "@uniswap/v4-core/contracts/libraries/Hooks.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {BalanceDelta} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+import {PoolIdLibrary} from "@uniswap/v4-core/contracts/types/PoolId.sol";
+import "../libraries/SwapIntention.sol";
+import {IQuoter} from "../interfaces/IQuoter.sol";
+import {PoolTicksCounter} from "../libraries/PoolTicksCounter.sol";
+
+contract Quoter is IQuoter {
+    using PoolIdLibrary for PoolKey;
+    using Hooks for IHooks;
+
+    /// @dev Transient storage variable used to check a safety condition in exact output swaps.
+    uint256 private amountOutCached;
+
+    // v4 Singleton contract
+    IPoolManager public immutable manager;
+
+    /// @dev function selector + length of bytes as uint256 + min length of revert reason padded to multiple of 32 bytes
+    uint256 internal constant MINIMUM_REASON_LENGTH = 68;
```

Comment:
Technically I think the min is 4 if it reverts with InvalidLockAcquiredSender or InvalidQuoteType

Reply:
Right now if that happens we will just bubble up UnexpectedRevertBytes. So would change this and then also add tests on these revert paths

Reply:
also this is minimum in the unhappy case right? so there is minimum valid and minimum invalid?

Reply:
added `uint256 internal constant MINIMUM_CUSTOM_ERROR_LENGTH = 4;`